zebra-state/src/error.rs

```diff
@@ -46,6 +46,18 @@ pub type BoxError = Box<dyn std::error::Error + Send + Sync + 'static>;
 #[error("block is not contextually valid: {}", .0)]
 pub struct CommitSemanticallyVerifiedError(#[from] ValidateContextError);
 
+/// An error describing the reason a block or its descendants could not be reconsidered after
+/// potentially being invalidated from the chain_set.
+#[derive(Debug, Error)]
+pub enum ReconsiderError {
+    #[error("Block with hash {0} was not previously invalidated")]
+    NonPreviouslyInvalidatedBlock(block::Hash),
```

Comment:
It would be nice to rename this error variant to `MissingInvalidatedBlock` instead for when blocks have been previously invalidated but were dropped from the cache of invalidated blocks.